api/src/main/java/marquez/common/Utils.java

```diff
@@ -172,4 +186,133 @@ public static Version newJobVersionFor(
             .getBytes(UTF_8);
     return Version.of(UUID.nameUUIDFromBytes(bytes));
   }
+
+  public static Version newDatasetVersionFor(
+      String namespace,
+      String sourceName,
+      String physicalName,
+      String datasetName,
+      List<LineageEvent.SchemaField> fields,
+      UUID runId) {
+    DatasetVersionData data =
+        DatasetVersionData.builder()
+            .namespace(namespace)
+            .sourceName(sourceName)
+            .physicalName(physicalName)
+            .datasetName(datasetName)
+            .schemaFields(fields)
+            .runId(runId)
+            .build();
+    return newDatasetVersionFor(data);
+  }
+
+  public static Version newDatasetVersionFor(
+      String namespaceName, String datasetName, DatasetMeta datasetMeta) {
+    DatasetVersionData datasetVersionData =
+        DatasetVersionData.builder()
+            .datasetName(datasetName)
+            .namespace(namespaceName)
+            .datasetMeta(datasetMeta)
+            .build();
+    return newDatasetVersionFor(datasetVersionData);
+  }
+
+  private static Version newDatasetVersionFor(DatasetVersionData data) {
+    if (data == null) return Version.of(UUID.nameUUIDFromBytes(("".getBytes())));
```

Comment:
When would the `DatasetVersionData` be **null**?

Reply:
In previous version it could be null, but after review, I've remove that option. It's never going to be null, so I've removed that checl